featurebyte/feast/patch.py

```diff
@@ -126,25 +158,20 @@ def get_transformed_features_df(
     pd.DataFrame
         Dataframe with transformed features
     """
-    # Apply on demand transformations
-    # Original implementation does not use dict to store the transformed features,
-    # which leads to pandas performance issues
-    columns_to_cleanup = []
-    df_dict = {}
+    # Original implementation assigns a new column on each iteration, this implementation uses a wrapper
+    # class to improve the runtime & memory performance.
+    df_with_features = DataFrameWrapper(df_with_features)
     for source_fv_projection in feature_view.source_feature_view_projections.values():
         for feature in source_fv_projection.features:
             full_feature_ref = f"{source_fv_projection.name}__{feature.name}"
             if full_feature_ref in df_with_features.keys():
                 # Make sure the partial feature name is always present
-                df_dict[feature.name] = df_with_features[full_feature_ref]
-                columns_to_cleanup.append(feature.name)
+                df_with_features.add_column_alias(full_feature_ref, feature.name)
             elif feature.name in df_with_features.keys():
                 # Make sure the full feature name is always present
-                df_dict[full_feature_ref] = df_with_features[feature.name]
-                columns_to_cleanup.append(full_feature_ref)
+                df_with_features.add_column_alias(feature.name, full_feature_ref)
 
     # Compute transformed values and apply to each result row
-    df_with_features = pd.concat([df_with_features, pd.DataFrame(df_dict)], axis=1)
```

Comment:
This statement takes about 30-40% runtime of the get online features for a feature list with 306 features.